scripts/pgpwave8_upgrade

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+usage()
+{
+    cat <<EOF
+usage $0 -i IOCNAME [-d DEVICE] [-r] [-p FIRMWARE_PATH]
+
+Read or upgrade the firmware version of a pgpwave8 given its ioc.
+
+-i The ioc whose pgpwave8's firmware should be upgraded
+-d The path to the device, defaulting to /dev/datadev_0
+-r Print the firmware version only. Cannot be used with -p
+-p The path to the mcs file containing the new firmware.
+Cannot be used with -r. Some firmware images can be found
+in /cds/group/pcds/package/wave8/images.
+EOF
+}
+
+remote()
+{
+    #!/bin/bash
+    # shellcheck disable=SC1091
+    source /cds/group/pcds/pyps/conda/rogue/etc/profile.d/conda.sh
+    if ! conda activate rogue_"$ROGUE_VERSION"; then
+        echo "A rogue environment for this version is not available."
+        exit 1
+    fi
+    if [[ ! -e $DEVICE ]]; then
+        echo "The device $DEVICE does not exist."
+        exit 1
+    fi
+    export PATH=/cds/group/pcds/engineering_tools/latest-released/scripts/:$PATH
+    # shellcheck disable=SC2034
+    if ! status=$(/cds/group/pcds/engineering_tools/latest-released/scripts/imgr "$IOCNAME" --status); then
+        echo "Unable to get the status of $IOCNAME. Exiting."
+        exit 1
+    fi
+
+    # change this to a common path later
+    WAVE8_REPO=/cds/home/t/tjohnson/trunk/workarea/wave8
+    if [[ $READ_VERSION -eq 1 ]]; then
+        if [[ $status == "RUNNING" ]]; then
+            pv=$(/cds/group/pcds/engineering_tools/latest-released/scripts/ioctool "$IOCNAME" pvs | grep FpgaVersion_RBV | awk '{print substr($1,1,length($1)-1)}')
+            caget "$pv" -0x
+        else
+	    # shellcheck disable=SC2164
+	    cd "$(dirname "$0")"
+            python pgpwave8_version.py -l "$LANE" --dev "$DEVICE"
+        fi
+        exit 0
```

Comment:
It feels strange to me to `exit` instead of `return` inside a function but maybe there's a good reason to do this for a remote script like this one is being used.